test/common/cro_cr_success.go

```diff
@@ -0,0 +1,128 @@
+package common
+
+import (
+	goctx "context"
+	"encoding/json"
+	"fmt"
+	crov1 "github.com/integr8ly/cloud-resource-operator/pkg/apis/integreatly/v1alpha1"
+	croTypes "github.com/integr8ly/cloud-resource-operator/pkg/apis/integreatly/v1alpha1/types"
+	integreatlyv1alpha1 "github.com/integr8ly/integreatly-operator/pkg/apis/integreatly/v1alpha1"
+	"github.com/integr8ly/integreatly-operator/pkg/resources/constants"
+	"k8s.io/apimachinery/pkg/types"
+	"testing"
+)
+
+const (
+	requestNameSpace  = "redhat-rhmi-operator"
+	requestUrl        = "/apis/integreatly.org/v1alpha1"
+	openShiftProvider = "openshift"
+	externalProvider  = "aws"
+)
+
+func TestCROPostGresSuccessfulState(t *testing.T, ctx *TestingContext) {
+	postGresToCheck := []string{
+		fmt.Sprintf("%s%s", constants.CodeReadyPostgresPrefix, InstallationName),
+		fmt.Sprintf("%s%s", constants.ThreeScalePostgresPrefix, InstallationName),
+		fmt.Sprintf("%s%s", constants.RHSSOPostgresPrefix, InstallationName),
+		fmt.Sprintf("%s%s", constants.RHSSOUserProstgresPrefix, InstallationName),
+		fmt.Sprintf("%s%s", constants.UPSPostgresPrefix, InstallationName),
+		constants.AMQAuthServicePostgres,
+	}
+
+	for _, postGresName := range postGresToCheck {
+
+		strategy := getResourceStrategy(t, ctx)
+		// AMQAuthService postgres is always in cluster
+		if postGresName == constants.AMQAuthServicePostgres {
+			strategy = openShiftProvider
+		}
+
+		requestBody, _ := getCustomResourceJson(t, ctx, "postgres", postGresName)
+
+		postGres := &crov1.Postgres{}
+		err := json.Unmarshal(requestBody, postGres)
+
+		if err != nil {
+			t.Fatalf("Failed to unmarshall json: %s", err)
+		}
```

Comment:
Wdyt about creating a function with this code? The steps in all three tests are always `getCustomResourceJson`, `json.Unmarshal`, and `if err` if I'm not wrong

Reply:
Yup, though the only difference is that we unmarshalling to difference CRO CR resources

Reply:
Yes, right the "Resource" or interface could be passed from outside like for the unmarshal function